src/circuit/layouter.rs

```diff
@@ -69,7 +69,7 @@ pub trait RegionLayouter<F: Field>: fmt::Debug {
         column: Column<Advice>,
         offset: usize,
         constant: Assigned<F>,
-    ) -> Result<Cell, Error>;
+    ) -> Result<AssignedCell<F, F>, Error>;
```

Comment:
This feels like an abstraction failure, because `RegionLayouter::assign_advice` and `RegionLayouter::assign_advice_from_instance` still return `Cell`. Either they should all return `AssignedCell`, or none of them should (and instead the caller should use `AssignedCell` constructors, as this PR currently provides for advice assigmnent).